conans/client/tools/scm.py

```diff
@@ -89,23 +99,48 @@ def run(self, command):
         command = self._configure_ssl_verify + command
         return super(Git, self).run(command)
 
-    def clone(self, url, branch=None, args=""):
+    def _fetch(self, url, branch, shallow):
+        if not branch:
+            raise ConanException("The destination folder '%s' is not empty, "
+                                 "specify a branch to checkout (not a tag or commit) "
+                                 "or specify a 'subfolder' "
+                                 "attribute in the 'scm'" % self.folder)
+
+        output = self.run("init")
+        output += self.run('remote add origin "%s"' % url)
```

Comment:
(Just thinking aloud:)

I know these are the same commands as before, but this command fails if the remote is already added to the repo:

```bash
⇒  git remote -v
origin	git@github.com:sword-and-sorcery/conanquest-fantastic-adventures.git (fetch)
origin	git@github.com:sword-and-sorcery/conanquest-fantastic-adventures.git (push)
[python3]jgsogo@MacBook-Pro-de-Javier:~/dev/conan/issues/tmp5514|master 
⇒  git remote add origin git@github.com:sword-and-sorcery/conanquest-fantastic-adventures.git
fatal: remoto origin ya existe.
```

Maybe it is just the name of the function, `fetch`, I know it is not public and won't be executed on its own, so these two lines probably will be removed in the future to the calling function.